polkadot/contracts/finance2/lib.rs

```diff
@@ -590,13 +598,16 @@ mod finance2 {
                 self.last_updated_at = updated_at;
                 
                 if let Some(c) = self.collateral.get(user) {
-                    let qouted_collateral = mulw(c, self.price).div(self.price_scaler).unwrap_or(u128::MAX);
-                    let icv = mulw(qouted_collateral, self.initial_haircut).scale();
-                    let mcv = mulw(qouted_collateral, self.maintenance_haircut).scale();
-                    Ok((self.next, 0, icv, 0, mcv, 0))
+                    if let Some(qouted_collateral) = mulw(c, self.price).div(self.price_scaler) {
+                        let icv = mulw(qouted_collateral, self.initial_haircut).scale();
+                        let mcv = mulw(qouted_collateral, self.maintenance_haircut).scale();
+                        Ok((self.next, 0, icv, 0, mcv, 0))
+                    } else {
+                        Ok((self.next, 0, u128::MAX, 0, u128::MAX, 0))
```

Comment:
The function repay_or_update must have a docstring describing what each of these return values represents